### PR TITLE
[ops] feat: add chunkloss characteristics for NPU

### PR DESCRIPTION
**Pull request by zhihaofang1017**
### What does this PR do?

Currently, large models on NPU may encounter high memory usage during training. 
This PR introduces ChunkLoss support to split the loss computation into smaller chunks, 
reducing peak memory usage without affecting training correctness.

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `misc`, `ci`, `config`, `docs`, `data`, `dist`, `omni`, `logging`, `model`, `optim`, `ckpt`, `release`, `task`, `perf`, `ops`, `parallel`
  - If this PR involves multiple modules, separate them with `,` like `[ci, data, model]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][parallel, model] feat: dynamic batching`

### Test
All changes have been validated with:
1. **Functional Tests**: Qwen3 Qwen3-VL passes inference/training correctness checks on Ascend NPU.
2. **Reduce memory**: The highest memory usage during training will be reduced about 20G
3. **Version Compatibility**: Passes tests on CANN(8.1.RC1~8.3.RC1)

**baseline**
image
```
Epoch 1/1:   0%|          | 1/3828049 [00:27<28735:05:00, 27.02s/it, loss: 2.3775, grad_norm: 5.3750, lr: 1.12e-08]
Epoch 1/1:   0%|          | 2/3828049 [00:47<24562:59:33, 23.10s/it, loss: 2.1726, grad_norm: 4.8125, lr: 2.24e-08]
Epoch 1/1:   0%|          | 3/3828049 [01:09<23983:52:54, 22.56s/it, loss: 2.2492, grad_norm: 5.8125, lr: 3.36e-08]
Epoch 1/1:   0%|          | 4/3828049 [01:30<23287:18:20, 21.90s/it, loss: 2.3213, grad_norm: 5.1250, lr: 4.48e-08]
Epoch 1/1:   0%|          | 5/3828049 [01:50<22693:03:30, 21.34s/it, loss: 2.2767, grad_norm: 5.1875, lr: 5.60e-08]
Epoch 1/1:   0%|          | 6/3828049 [02:11<22467:07:23, 21.13s/it, loss: 2.3104, grad_norm: 4.6875, lr: 6.72e-08]
Epoch 1/1:   0%|          | 7/3828049 [02:31<22178:22:12, 20.86s/it, loss: 2.1088, grad_norm: 5.4062, lr: 7.84e-08]
Epoch 1/1:   0%|          | 8/3828049 [02:51<22018:18:56, 20.71s/it, loss: 2.1520, grad_norm: 5.7188, lr: 8.96e-08]
Epoch 1/1:   0%|          | 9/3828049 [03:12<21941:05:21, 20.63s/it, loss: 2.3253, grad_norm: 5.1250, lr: 1.01e-07]
Epoch 1/1:   0%|          | 10/3828049 [03:32<21874:42:09, 20.57s/it, loss: 2.3602, grad_norm: 4.7188, lr: 1.12e-07]
Epoch 1/1:   0%|          | 11/3828049 [03:53<22019:28:31, 20.71s/it, loss: 2.3888, grad_norm: 5.0938, lr: 1.23e-07]
Epoch 1/1:   0%|          | 12/3828049 [04:14<21887:35:51, 20.58s/it, loss: 2.3280, grad_norm: 4.8438, lr: 1.34e-07]
Epoch 1/1:   0%|          | 13/3828049 [04:35<21991:36:28, 20.68s/it, loss: 2.1234, grad_norm: 5.9688, lr: 1.46e-07]
Epoch 1/1:   0%|          | 14/3828049 [04:55<21903:59:18, 20.60s/it, loss: 2.1497, grad_norm: 4.8438, lr: 1.57e-07]
Epoch 1/1:   0%|          | 15/3828049 [05:16<21923:21:32, 20.62s/it, loss: 2.2755, grad_norm: 4.8125, lr: 1.68e-07]
Epoch 1/1:   0%|          | 16/3828049 [05:36<21850:10:35, 20.55s/it, loss: 2.2511, grad_norm: 5.1875, lr: 1.79e-07]
Epoch 1/1:   0%|          | 17/3828049 [05:57<21847:57:17, 20.55s/it, loss: 2.1966, grad_norm: 5.1875, lr: 1.90e-07]
Epoch 1/1:   0%|          | 18/3828049 [06:17<21752:44:07, 20.46s/it, loss: 2.2671, grad_norm: 5.2188, lr: 2.02e-07]
Epoch 1/1:   0%|          | 19/3828049 [06:37<21825:10:46, 20.53s/it, loss: 2.2315, grad_norm: 5.4375, lr: 2.13e-07]
Epoch 1/1:   0%|          | 20/3828049 [06:58<21857:18:12, 20.56s/it, loss: 2.2359, grad_norm: 5.1875, lr: 2.24e-07]
```

**chunkloss**
image

```
Epoch 1/1:   0%|          | 1/3828049 [00:26<28408:06:59, 26.72s/it, loss: 2.3766, grad_norm: 5.3750, lr: 1.12e-08]
Epoch 1/1:   0%|          | 2/3828049 [00:47<24402:28:47, 22.95s/it, loss: 2.1726, grad_norm: 4.8125, lr: 2.24e-08]
Epoch 1/1:   0%|          | 3/3828049 [01:09<24035:08:06, 22.60s/it, loss: 2.2492, grad_norm: 5.8125, lr: 3.36e-08]
Epoch 1/1:   0%|          | 4/3828049 [01:30<23424:54:35, 22.03s/it, loss: 2.3222, grad_norm: 5.1875, lr: 4.48e-08]
Epoch 1/1:   0%|          | 5/3828049 [01:50<22786:20:33, 21.43s/it, loss: 2.2769, grad_norm: 5.2188, lr: 5.60e-08]
Epoch 1/1:   0%|          | 6/3828049 [02:11<22510:32:53, 21.17s/it, loss: 2.3105, grad_norm: 4.6875, lr: 6.72e-08]
Epoch 1/1:   0%|          | 7/3828049 [02:31<22209:17:14, 20.89s/it, loss: 2.1079, grad_norm: 5.4062, lr: 7.84e-08]
Epoch 1/1:   0%|          | 8/3828049 [02:52<22028:57:10, 20.72s/it, loss: 2.1529, grad_norm: 5.7188, lr: 8.96e-08]
Epoch 1/1:   0%|          | 9/3828049 [03:12<21876:17:27, 20.57s/it, loss: 2.3244, grad_norm: 5.1250, lr: 1.01e-07]
Epoch 1/1:   0%|          | 10/3828049 [03:32<21816:16:37, 20.52s/it, loss: 2.3605, grad_norm: 4.7500, lr: 1.12e-07]
Epoch 1/1:   0%|          | 11/3828049 [03:53<21916:52:51, 20.61s/it, loss: 2.3890, grad_norm: 5.0938, lr: 1.23e-07]
Epoch 1/1:   0%|          | 12/3828049 [04:13<21799:11:21, 20.50s/it, loss: 2.3285, grad_norm: 4.8438, lr: 1.34e-07]
Epoch 1/1:   0%|          | 13/3828049 [04:34<21917:16:46, 20.61s/it, loss: 2.1227, grad_norm: 5.6875, lr: 1.46e-07]
Epoch 1/1:   0%|          | 14/3828049 [04:55<21871:20:57, 20.57s/it, loss: 2.1498, grad_norm: 4.8438, lr: 1.57e-07]
Epoch 1/1:   0%|          | 15/3828049 [05:15<21967:32:05, 20.66s/it, loss: 2.2754, grad_norm: 4.8125, lr: 1.68e-07]
Epoch 1/1:   0%|          | 16/3828049 [05:36<21885:34:18, 20.58s/it, loss: 2.2512, grad_norm: 5.1875, lr: 1.79e-07]
Epoch 1/1:   0%|          | 17/3828049 [05:57<21966:24:05, 20.66s/it, loss: 2.1969, grad_norm: 5.1875, lr: 1.90e-07]
Epoch 1/1:   0%|          | 18/3828049 [06:17<21815:08:13, 20.52s/it, loss: 2.2672, grad_norm: 5.2188, lr: 2.02e-07]
Epoch 1/1:   0%|          | 19/3828049 [06:38<21915:39:06, 20.61s/it, loss: 2.2318, grad_norm: 5.4375, lr: 2.13e-07]
Epoch 1/1:   0%|          | 20/3828049 [06:58<21922:43:07, 20.62s/it, loss: 2.2361, grad_norm: 5.1875, lr: 2.24e-07]
Epoch 1/1:   0%|          | 21/3828049 [07:19<22083:13:38, 20.77s/it, loss: 2.0880, grad_norm: 4.3750, lr: 2.35e-07]
Epoch 1/1:   0%|          | 22/3828049 [07:40<21966:00:46, 20.66s/it, loss: 2.7869, grad_norm: 6.1250, lr: 2.46e-07]
```

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md?plain=1#L22): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/ByteDance-Seed/VeOmni/blob/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/ByteDance-Seed/VeOmni/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...

### CODE REVIEW

File: veomni/ops/fused_cross_entropy/__init__.py [hunk 1]
```diff
@@ -103,9 +104,100 @@ def ForCausalLMLoss(
...
+    return chunk_loss, None
+
+
 def apply_veomni_loss_patch():
     LOSS_MAPPING["ForCausalLM"] = ForCausalLMLoss
-    LOSS_MAPPING["ForConditionalGeneration"] = ForCausalLMLoss
+    LOSS_MAPPING["ChunkLoss"] = chunk_loss_function
     global _cross_entropy
     if is_torch_npu_available():
         _cross_entropy = eager_cross_entropy
```
Comment (onehaitao) [inline veomni/ops/fused_cross_entropy/__init__.py:205]: I think you can update` LOSS_MAPPING["ForCausalLM"]` and `LOSS_MAPPING["ForConditionalGeneration"]` here with `chunk_loss_function`, and maintain the original logic.

Comment (zhihaofang1017) [reply veomni/ops/fused_cross_entropy/__init__.py:205]: I have fixed it and move chunkloss into is_torch_npu_available()

File: veomni/ops/fused_cross_entropy/__init__.py [hunk 2]
```diff
@@ -103,11 +104,104 @@ def ForCausalLMLoss(
...
+            num_items_in_batch,
+            ignore_index,
+            shift_labels,
+            hidden_states=hidden_states,
+            weights=weights,
+            **kwargs,
+        )
+        return loss, logits
+
+    chunk_size = 1024
```
Comment (FoolPlayer) [inline veomni/ops/fused_cross_entropy/__init__.py]: Is 1024 a fixed size or will be changed in some case?

File: veomni/ops/fused_cross_entropy/__init__.py [hunk 3]
```diff
@@ -103,9 +104,100 @@ def ForCausalLMLoss(
...
+    chunk_labels = torch.split(labels, chunk_size, dim=1)
+
+    loss_kwargs_chunks = [
+        {"labels": chunk_labels[i], "ignore_index": -100, "num_items_in_batch": (labels != ignore_index).sum()}
+        for i in range(len(chunk_labels))
+    ]
```
[the remark this answers is not in the document]
Comment (FoolPlayer) [reply veomni/ops/fused_cross_entropy/__init__.py:192]: Make sense, it's better to use the function args `ignore_index`

File: docs/get_started/installation/install_ascend.md [hunk 4]
```diff
@@ -44,4 +44,7 @@ pip3 install -e .[npu]
 # Make sure CANN_path is set to your CANN installation directory, e.g., export CANN_path=/usr/local/Ascend
 source $CANN_path/ascend-toolkit/set_env.sh
 export PYTORCH_NPU_ALLOC_CONF=expandable_segments:True
-```
+
+# Add chunkloss feature
+export VEOMNI_ENABLE_CHUNK_LOSS=1
```
Comment (FoolPlayer) [inline docs/get_started/installation/install_ascend.md:49]: Maybe we don't need this env, like liger-kernel the patch for cross_entropy loss fn is default.

Comment (zhihaofang1017) [reply docs/get_started/installation/install_ascend.md:49]: Let's add a switch for now, since we haven't validated other models yet.

File: veomni/ops/fused_cross_entropy/__init__.py (hunk 2 above)
Comment (zhihaofang1017) [reply veomni/ops/fused_cross_entropy/__init__.py]: This is a  fixed size

Comment (zhihaofang1017) [reply veomni/ops/fused_cross_entropy/__init__.py]: I have added chunksize as a args
